lib/cocoapods/sandbox/file_accessor.rb

```diff
@@ -305,17 +305,36 @@ def docs
         path_list.glob([GLOB_PATTERNS[:docs]])
       end
 
+      # Validates that the podspec's `license` file exists if a path is specified
+      # Returns the pathname if found and the path exists
+      #
+      # @return [Pathname] the path of the license file, if specified and valid
+      #
+      def validate_spec_license!
```

Comment:
Any reason this method name includes `!`? I am not a ruby expert, but I am under the impression that using `!` means that the method will alter state of an object. Perhaps maybe because it might also crash? 

Neither of the above appear to be true I think here (besides the warning).

Also I think the word `validate` is a bit misleading here as this method should probably be `void` versus this one actually returning a `Pathname` here.

Are both of these questions a pattern in this file?

Reply:
Ah, sorry that's my lack of experience in ruby showing. My main concern here was that I wanted a method which would return the Pathname if it exists and is valid, but also will print a warning if the path is missing. I can move this logic back to inside `developer_files` if that is preferred